libmobilecoin/src/chacha20_rng.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use crate::common::*;
+use mc_util_ffi::*;
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+use std::{convert::TryInto, sync::Mutex};
+
+pub type McChaCha20Rng = ChaCha20Rng;
+
+// McU128 facilitates conversion between u128 and 16 bytes of u8
+pub struct McU128 {
+    pub bytes: [u8; 16],
+}
+
+impl McU128 {
+    pub fn from_u128(val: u128) -> McU128 {
+        McU128 {
+            bytes: val.to_be_bytes(),
+        }
+    }
+
+    pub fn to_u128(&self) -> u128 {
+        u128::from_be_bytes(self.bytes)
+    }
+}
+
+impl IntoFfi<McU128> for McU128 {
+    #[inline]
+    fn error_value() -> McU128 {
+        McU128 {
+            bytes: [u8::MAX; 16],
+        }
+    }
+
+    #[inline]
+    fn into_ffi(self) -> McU128 {
+        self
+    }
+}
+
+impl_into_ffi!(FfiOwnedPtr<McU128>);
```

Comment:
Is this actually necessary? Can this just be Rust's u128?

Reply:
I was thinking the same thing and was going to try to clean this out.  Originally I had used McU128 as a parameter type, but started pulling it out to get rid of it - but still had it here to convert to/from [u8; 16]

Reply:
Removed - thanks!